include/misc/mempool_base.h

```diff
@@ -36,10 +36,12 @@ struct sys_mem_pool_base {
 	u8_t flags;
 };
 
-#define Z_MPOOL_HAVE_LVL(maxsz, minsz, l) (((maxsz) >> (2*(l))) \
-					  >= (minsz) ? 1 : 0)
+#define _MPOOL_MINBLK sizeof(sys_dnode_t)
+
+#define Z_MPOOL_HAVE_LVL(maxsz, minsz, l) \
+	(((maxsz) >> (2*(l))) >= MAX(minsz, _MPOOL_MINBLK) ? 1 : 0)
```

Comment:
I'm having trouble understanding this patch too.  This description seems wrong to me:

> The problem comes from Z_MPOOL_LVLS() which returns the number of levels
being tested against the leven number with >= and being off by one level
(if we have 2 levels, the test should be true for level 0 and 1 only).

Consider: we have a heap with a maxsz of 64 and a minsz of 16.  The top block will split into four 16 byte subblocks cleanly, but those blocks are now at the minimum size and won't split any further.  So we have levels zero and one, and no others, for a total of two.   The logic in Z_MPOOL_HAVE_LVL() seems to match those semantics already, and a quick test shows that it's true at runtime as expected and __MPOOL_LVLS(64, 16) evaluates to two.

If there's waste, then it's not from this particular issue, I think.

> This could be simply fixed by turning the >= into  > but [...]

So.... I know you can hear this coming, but could you please submit that one byte patch with a fixed explanation instead?